src/corsEventHandler.ts

```diff
@@ -6,11 +6,20 @@
 
 /*****************************************************************************************************************/
 
-import { type EventHandler, type H3CorsOptions, defineEventHandler, handleCors } from 'h3'
+import {
+  type EventHandler,
+  type EventHandlerRequest,
+  type H3CorsOptions,
+  defineEventHandler,
+  handleCors
+} from 'h3'
 
 /*****************************************************************************************************************/
 
-export const corsEventHandler = <T>(handler: EventHandler<T>, options: H3CorsOptions) => {
+export const corsEventHandler = <T extends EventHandlerRequest>(
```

Comment:
we now have a second generic which is the response type, so we likely want to pass that one on as well.

Reply:
@danielroe Good shout. Have amended.